### PR TITLE
Create helper functions to get CPU cores info on linux

### PR DESCRIPTION
**Pull request by akindyakov**
Helper functions to get CPU cores info exposed via sysfs on linux.
I need this information to use `kprobes` in osquery on linux.

### CODE REVIEW

File: osquery/core/linux/cpu.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ *  This source code is licensed as defined on the LICENSE file found in the
+ *  root directory of this source tree.
+ */
+
+#pragma once
+
+#include <osquery/expected.h>
+
+#include <string>
+#include <vector>
```
Comment (guliashvili) [inline osquery/core/linux/cpu.h]: not used?

File: osquery/core/linux/cpu.cpp [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+} // namespace
+
+Expected<Mask, Error> decodeMaskFromString(const std::string& encoded_str) {
+  auto mask = Mask{};
+  if (encoded_str.empty()) {
+    return mask;
+  }
+  using It = boost::split_iterator<std::string::const_iterator>;
+  for (auto it = boost::make_split_iterator(
```
Comment (guliashvili) [inline osquery/core/linux/cpu.cpp]: const ..& ?

File: osquery/core/linux/cpu.cpp [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+#include "osquery/core/linux/cpu.h"
+
+#include <osquery/core/conversions.h>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/io/detail/quoted_manip.hpp>
+
+#include <fstream>
+#include <functional>
```
Comment (guliashvili) [inline osquery/core/linux/cpu.cpp]: not used?

File: osquery/core/linux/cpu.cpp (hunk 2 above)
Comment (akindyakov) [reply osquery/core/linux/cpu.cpp]: It is an iterator, it have to be copied and incremented at the every step of the loop.
However it is a `const_iterator`, that means our data is in safe ;)

File: osquery/core/linux/cpu.h (hunk 1 above)
Comment (akindyakov) [reply osquery/core/linux/cpu.h]: Yes =\ it is supposed to be `bitset`, thanks!

File: osquery/core/linux/cpu.cpp (hunk 3 above)
Comment (akindyakov) [reply osquery/core/linux/cpu.cpp]: yep, thanks!

**Comment by akindyakov** [comment]
Thank you @guliashvili , for you review!

**Pull request merged**